number_theory/modular.h

```
@@ -46,6 +48,27 @@ struct ModulusWrapper {
 
 }  // namespace modular_internal
 
+// Returns the modular inverse of |number| in modulo |mod| if exists.
+// Otherwise, throws an invalid_argument exception.
+template <typename T>
+T inverse_mod(const T &number, const T &mod) {
+  static_assert(std::numeric_limits<T>::is_integer);
```

I think we need to throw an invalid_argument exception when mod<=0.